### PR TITLE
Add jupyter-compatible repr for SerializableDevice

### PR DESCRIPTION
**Pull request by maffoo**
Makes working with SerializableDevice in ipython or jupyter a bit nicer.

Before:
```
In [1]: import cirq                                                                                      

In [2]: cirq.google.Sycamore                                                                             
Out[2]: <cirq.google.devices.serializable_device.SerializableDevice at 0x7fa7e2840b50>
```

After:
```
In [1]: import cirq                                                                                      

In [2]: cirq.google.Sycamore                                                                             
Out[2]: 
                                             (0, 5)───(0, 6)
                                             │        │
                                             │        │
                                    (1, 4)───(1, 5)───(1, 6)───(1, 7)
                                    │        │        │        │
                                    │        │        │        │
                           (2, 3)───(2, 4)───(2, 5)───(2, 6)───(2, 7)───(2, 8)
                           │        │        │        │        │        │
                           │        │        │        │        │        │
                  (3, 2)───(3, 3)───(3, 4)───(3, 5)───(3, 6)───(3, 7)───(3, 8)───(3, 9)
                  │        │        │        │        │        │        │        │
                  │        │        │        │        │        │        │        │
         (4, 1)───(4, 2)───(4, 3)───(4, 4)───(4, 5)───(4, 6)───(4, 7)───(4, 8)───(4, 9)
         │        │        │        │        │        │        │        │
         │        │        │        │        │        │        │        │
(5, 0)───(5, 1)───(5, 2)───(5, 3)───(5, 4)───(5, 5)───(5, 6)───(5, 7)───(5, 8)
         │        │        │        │        │        │        │
         │        │        │        │        │        │        │
         (6, 1)───(6, 2)───(6, 3)───(6, 4)───(6, 5)───(6, 6)───(6, 7)
                  │        │        │        │        │
                  │        │        │        │        │
                  (7, 2)───(7, 3)───(7, 4)───(7, 5)───(7, 6)
                           │        │        │
                           │        │        │
                           (8, 3)───(8, 4)───(8, 5)
                                    │
                                    │
                                    (9, 4)
```

### CODE REVIEW

File: cirq/google/devices/serializable_device_test.py [hunk 1]
```diff
@@ -50,6 +52,14 @@
 )
 
 
+@pytest.mark.parametrize('cycle,func', [(False, str), (True, repr)])
+def test_repr_pretty(cycle, func):
+    device = cg.Sycamore
+    printer = mock.Mock()
+    device._repr_pretty_(printer, cycle)
```
Comment (dstrain115) [inline cirq/google/devices/serializable_device_test.py:59]: Doesn't this test basically just call the function and assert that it was called?

I was thinking we should test that the grid gets printed out?

Comment (viathor) [reply cirq/google/devices/serializable_device_test.py:59]: Well, we call one function and assert that *another* function was called. This is called interaction testing. It is generally considered a bit harder to maintain (e.g. a refactoring that yields the same result using a different interaction will need to modify the test), so +1 to Doug's suggestion.

Comment (maffoo) [reply cirq/google/devices/serializable_device_test.py:59]: TBH I mostly added this test so the coverage checks wouldn't complain at me :-)

But I will add a separate test checking that `SerializableDevice.__str__` outputs a nice grid.

Comment (balopat) [reply cirq/google/devices/serializable_device_test.py:59]: +1 on not relying on interaction testing / mocks but rather actual behavior/output.

**Comment by maffoo** [comment]
Build docs action failing to install `pyscf`:
```
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    pyscf; sys_platform != "win32" from https://files.pythonhosted.org/packages/b0/d0/9858aa09bc330c31025be64c385881403cbe178c44bf6fb5880bd84b3639/pyscf-1.7.4-cp36-cp36m-manylinux1_x86_64.whl#sha256=0877719b64ef6081b63e2f6c1cea8f78661ab2a835aaa302a12ce12ff4411539 (from qiskit-aqua==0.6.1->qiskit~=0.13.0->-r dev_tools/conf/pip-list-dev-tools.txt (line 21)):
        Expected sha256 0877719b64ef6081b63e2f6c1cea8f78661ab2a835aaa302a12ce12ff4411539
             Got        928387c224fbdc060c240202ffe98695083e5391e691b6f6a7b299119e17bcbd
```
@balopat, is this a known issue?

**Comment by balopat** [comment]
> Build docs action failing to install `pyscf`:
> 
> ```
> ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
>     pyscf; sys_platform != "win32" from https://files.pythonhosted.org/packages/b0/d0/9858aa09bc330c31025be64c385881403cbe178c44bf6fb5880bd84b3639/pyscf-1.7.4-cp36-cp36m-manylinux1_x86_64.whl#sha256=0877719b64ef6081b63e2f6c1cea8f78661ab2a835aaa302a12ce12ff4411539 (from qiskit-aqua==0.6.1->qiskit~=0.13.0->-r dev_tools/conf/pip-list-dev-tools.txt (line 21)):
>         Expected sha256 0877719b64ef6081b63e2f6c1cea8f78661ab2a835aaa302a12ce12ff4411539
>              Got        928387c224fbdc060c240202ffe98695083e5391e691b6f6a7b299119e17bcbd
> ```
> 
> @balopat, is this a known issue?

Haven't seen that one before. But it seems to be passing now - I'll open a flake issue for it just in case we see it again, maybe we can reproduce it / understand it better.

**Pull request merged**